### PR TITLE
Fix archive.replacements in goreleaser

### PR DESCRIPTION
**Pull request by eapolinario**
# TL;DR
Similar to https://github.com/flyteorg/flytepropeller/pull/586/

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 Same fix as https://github.com/flyteorg/flytepropeller/pull/586/.

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -31,17 +31,12 @@ archives:
   - id: flyteadmin-archive
     name_template: |-
       flyteadmin_{{ .Tag }}_{{ .Os }}_{{ .Arch -}}
```
Comment (jeevb) [inline .goreleaser.yml:33]:
```suggestion
      flyteadmin_{{ .Tag }}_{{ .Os }}_
```

**Review by jeevb — APPROVED** [review]

**Review by pingsutw — APPROVED** [review]
